### PR TITLE
feat(RHTAPWATCH-820): Document the Project-controller

### PR DESCRIPTION
**Pull request by ifireball**
Document how to use the project-controller which helps with managing multiple software versions.

### CODE REVIEW

File: docs/modules/ROOT/pages/how-to-guides/proc_multiversion.adoc [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+The Konflux Project Controller seeks to streamline the process of managing
+multiple versions by introducing the following objects and concepts:
+
+- A *Project* is used to describe a major piece of software that can be worked
+  on by multiple teams over an extended period of time. A project may contain
+  one or more development streams.
+- A *ProjectDevelopmentStream* indicates an independent stream of development.
+  A *ProjectDevelopmentStream* can contain one or more *Applications* each
+  containing one or more *Components*.
+- A described above, starting a new development stream involves creating a large
```
Comment (yftacherzog) [inline docs/modules/ROOT/pages/how-to-guides/proc_multiversion.adoc:20]:
```suggestion
- As described above, starting a new development stream involves creating a large
```

File: docs/modules/ROOT/pages/how-to-guides/proc_multiversion.adoc [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+
+== Using the Project Controller
+
+In the sections below we will describe how to use the custom resources
+supported by the Project Controller to manage multiple software versions and
+generate Application and Component resources.
+
+=== Before you begin
+
+Its possible to have the Project Controller create all the Application and
```
Comment (yftacherzog) [inline docs/modules/ROOT/pages/how-to-guides/proc_multiversion.adoc:35]:
```suggestion
It's possible to have the Project Controller create all the Application and
```

File: docs/modules/ROOT/pages/how-to-guides/proc_multiversion.adoc [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+Component resources you would need. It is recommended, however, for the first
+time a particular repository is on-boarded, to do so manually via the Konflux
+UI. That would help ensuring the right access credentials are in place as well
+as that the Tekton Pipeline-as-Code files are committed to the repo.
+
+When creating components the Project Controller assumes that:
+
+- The default pipeline is not being used and that each branch in a component's
+  repo contains the necessary Tekton Pipeline-as-Code files to build it.
+- Any credentials required to access the component repo had already bee
```
Comment (yftacherzog) [inline docs/modules/ROOT/pages/how-to-guides/proc_multiversion.adoc:45]:
```suggestion
- Any credentials required to access the component repo had already been
```

File: docs/modules/ROOT/pages/how-to-guides/proc_multiversion.adoc [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+** The `spec.containerImage` field should be removed.
+** The following fields should probably contain variable references:
+*** `spec.application`
+*** `spec.componentName`
+*** `source.git.revision`
+
+Some notes about using template variables:
+
+* You can use the https://pkg.go.dev/text/template[Go text/template] syntax to
+  place template variable  values into various resource attributes as well as
```
Comment (yftacherzog) [inline docs/modules/ROOT/pages/how-to-guides/proc_multiversion.adoc:148]:
```suggestion
  place template variable values into various resource attributes as well as
```

File: docs/modules/ROOT/pages/how-to-guides/proc_multiversion.adoc [hunk 5]
```diff
@@ -0,0 +1,236 @@
...
+  variable default values.
+* You can use the custom `hyphenize` template function to create a value
+  suitable for use in resource names.
+* Its a advisable to quote strings that contain variable references and other
```
Comment (yftacherzog) [inline docs/modules/ROOT/pages/how-to-guides/proc_multiversion.adoc:152]:
```suggestion
* It's advisable to quote strings that contain variable references and other
```

**Review by yftacherzog — COMMENTED** [review]
lgtm

**Review by Chr1st1anSears — APPROVED** [review]

**Comment by ifireball** [comment]
@Chr1st1anSears given that the controller described by this PR is already deployed to production, could you merge this so people can lean how to use it?

**Comment by Chr1st1anSears** [comment]
> @Chr1st1anSears given that the controller described by this PR is already deployed to production, could you merge this so people can lean how to use it?

Just merged--fyi, I approved the changes so you could merge it. In the future, please feel free to pull the trigger :+1:

**Comment by ifireball** [comment]
> > @Chr1st1anSears given that the controller described by this PR is already deployed to production, could you merge this so people can lean how to use it?
> 
> Just merged--fyi, I approved the changes so you could merge it. In the future, please feel free to pull the trigger 👍

@Chr1st1anSears if I could have I would have....

**Pull request merged**